### PR TITLE
Bug 1219628 - Test Browser Save Image r=jlorenzo

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d31f10691bbff7e75fa6cd4d1727cf7ac0fb1376

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0e39bd41760580d1dbeca719de8fa275d29a7310

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2fb6ea4b16ae468b767f90645a0081a9cc33ef41

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=35943f100a35b988e764f9eb81c58edc9d3b9d4f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=57cef4e405e3ab8666b686effd30cfa742375191

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=778e29747ecea7ccdd805ba6cce3ef104b5bdcbc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=117345a89aa3813bac106c1be866cf867fe89854

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=85ad9ef195146abf9cfa69044255c002fffeb752

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=36ca71590edec680efaa09e041de6aaa8fb50b90

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=237498bd5075df42c6c1967ac150501ff5ffb144

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=112ea4cfc6c53bd458f8b8da49641026e76f1a6a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=112ea4cfc6c53bd458f8b8da49641026e76f1a6a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4a5ae4fa5737ad8036a2b14d836f0e934ccc7941

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6c8e4dddf912b1c82d9fdbbdafe791213bbcf0ce

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8c36da9d0058a2b99b82e35bf320488b25df3963

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1cd4700463bcd466a76d11b816459f5d58f59d2a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b129959ce6225dc5bc9e8ca475b8662bae3e8dea

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b48aed25b9c69e7597c0b59d78b96feeb1a22ae8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=46a063a13be916734f8d2da8c72ab5b39e85cc9b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ccb22dcef87b596c9738b1e6f6868b85212dd7aa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=810a4978a60395ca1605328b9afe28fb32e8961b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3660e0d335271c8c60f762d88370f80f29aac6f4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=891a1dbf3a3d9c1c949e855ef17975f49435725b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6ab09d63ca43777e687be60612d93280c772de67

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=450e67778e535d485aa9746edf562029433b080d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b217bcd936661d8fd4328e7fb69c6a975f8fe43

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0d6488ae579f0169bba482336570a00b800c7cb3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=860c6a2bdc7bdaf609b7dc5c1cdd7899636e2bf9

File: apps/system/test/marionette/lib/system.js [hunk 1]
```diff
@@ -363,6 +380,34 @@ System.prototype = {
     this.waitForActivityMenu();
   },
 
+  saveImage: function() {
+    this.client.switchToFrame();
+    this.client.waitFor(function() {
```
Comment (mikehenrty) [inline apps/system/test/marionette/lib/system.js]: nit: no need to do this waitFor, `this.appChromeContextMenu.displayed()` internally does a `waitForElement` on the context menu

File: apps/system/test/marionette/browser_context_menu_test.js [hunk 2]
```diff
@@ -39,11 +39,36 @@ marionette('Browser Context Menu', function() {
...
+    var url = server.url('sample_image.html');
+    rocketbar.goToURL(url);
+    system.gotoBrowser(url);
+
+    var image = client.helper.waitForElement('img');
+    actions.longPress(image, 0.5).perform(function() {
+      system.saveImage();
+
+      client.switchToFrame();
+      client.waitFor(function() {
```
Comment (mikehenrty) [inline apps/system/test/marionette/browser_context_menu_test.js]: nit: no need to do this wait for `system.toaster.display()` will internally do a `waitForElement`.

File: apps/system/test/marionette/browser_context_menu_test.js [hunk 3]
```diff
@@ -39,11 +39,36 @@ marionette('Browser Context Menu', function() {
...
+        return system.toaster.displayed();
+      });
+
+      client.waitFor(function() {
+        return !system.toaster.displayed();
+      });
```
Comment (mikehenrty) [inline apps/system/test/marionette/browser_context_menu_test.js]: these lines starting with 58 could be shortened to:
```
var toaster = system.toaster;
client.helper.waitForElementToDisappear(toaster);

File: apps/system/test/marionette/lib/system.js [hunk 4]
```diff
@@ -300,6 +307,16 @@ System.prototype = {
     return this.client.findElement(System.Selector.utilityTrayMotion);
   },
 
+  get toaster() {
+    var toaster;
+    this.client.waitFor(function() {
+      toaster = this.client.findElement(System.Selector.toaster);
+      return toaster ? true : false;
+    }.bind(this));
```
Comment (mikehenrty) [inline apps/system/test/marionette/lib/system.js]: Ahhh, I didn't realize this didn't do a `waitForElement(System.Selector.toaster)`. If you changed it to do that instead, you could simplify the logic above that waits for it to appear and then disappear.

**Pull request merged**